# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-tag-attributes-private.h
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-tag-attributes-private.h
--- a/gfx/cairo/cairo/src/cairo-tag-attributes-private.h	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-tag-attributes-private.h	Mon May 06 19:25:12 2024 +0000
@@ -49,6 +49,15 @@
     TAG_LINK_FILE,
 } cairo_tag_link_type_t;
 
+typedef struct _cairo_content_attrs {
+    char *id;
+    char *tag_name;
+} cairo_content_attrs_t;
+
+typedef struct _cairo_content_ref_attrs {
+    char *ref;
+} cairo_content_ref_attrs_t;
+
 typedef struct _cairo_link_attrs {
     cairo_tag_link_type_t link_type;
     cairo_array_t rects;
@@ -58,6 +67,9 @@
     int page;
     cairo_bool_t has_pos;
     cairo_point_double_t pos;
+    char *id;
+    char *ref;
+    int link_page;
 } cairo_link_attrs_t;
 
 typedef struct _cairo_dest_attrs {
@@ -91,9 +103,27 @@
 _cairo_tag_parse_dest_attributes (const char *attributes, cairo_dest_attrs_t *dest_attrs);
 
 cairo_private cairo_int_status_t
+_cairo_tag_parse_content_attributes (const char *attributes, cairo_content_attrs_t *content_attrs);
+
+cairo_private cairo_int_status_t
+_cairo_tag_parse_content_ref_attributes (const char *attributes, cairo_content_ref_attrs_t *content_ref_attrs);
+
+cairo_private cairo_int_status_t
 _cairo_tag_parse_ccitt_params (const char *attributes, cairo_ccitt_params_t *dest_attrs);
 
 cairo_private cairo_int_status_t
 _cairo_tag_parse_eps_params (const char *attributes, cairo_eps_params_t *dest_attrs);
 
+cairo_private void
+_cairo_tag_free_link_attributes (cairo_link_attrs_t *link_attrs);
+
+cairo_private void
+_cairo_tag_free_dest_attributes (cairo_dest_attrs_t *link_attrs);
+
+cairo_private void
+_cairo_tag_free_content_attributes (cairo_content_attrs_t *link_attrs);
+
+cairo_private void
+_cairo_tag_free_content_ref_attributes (cairo_content_ref_attrs_t *link_attrs);
+
 #endif /* CAIRO_TAG_ATTRIBUTES_PRIVATE_H */